python/llm/src/bigdl/llm/transformers/modelling_bigdl.py

```diff
@@ -20,58 +20,209 @@
 
 from bigdl.llm.utils.common import invalidInputError
+from .model import AutoModelForCausalLM
 
 
-class BigdlNativeForCausalLM:
-    """
-    A generic model class that mimics the behavior of
-    ``transformers.LlamaForCausalLM.from_pretrained`` API
-    """
+class LlamaForCausalLM:
 
     @classmethod
     def from_pretrained(cls,
                         pretrained_model_name_or_path: str,
-                        model_family: str = 'llama',
-                        dtype: str = 'int4',
+                        native: bool = True,
+                        dtype: str = "int4",
+                        *args,
                         **kwargs):
         """
-        :param pretrained_model_name_or_path: Path for converted BigDL-LLM optimized ggml
-               binary checkpoint. The checkpoint should be converted by ``bigdl.llm.llm_convert``.
-        :param model_family: The model family of the pretrained checkpoint.
-               Currently we support ``"llama"``, ``"bloom"``, ``"gptneox"``, ``"starcoder"``
-               and ``"chatglm"``.
+        :param pretrained_model_name_or_path: Path for model checkpoint.
+               If running with ``native int4``, the path should be converted BigDL-LLM optimized
+               ggml binary checkpoint, which should be converted by ``bigdl.llm.llm_convert``.
+               If running with ``transformers int4``, the path should be the huggingface repo id
+               to be downloaded or the huggingface checkpoint folder.
+        :param native: Load model to either BigDL-LLM optimized Transformer or Native (ggml) int4.
         :param dtype: Which quantized precision will be converted.
-                Now only `int4` and `int8` are supported, and `int8` only works for `llama`
-                , `gptneox` and `starcoder`.
-        :param cache_dir: (optional) This parameter will only be used when
-               ``pretrained_model_name_or_path`` is a huggingface checkpoint or hub repo id.
+               Now only `int4` and `int8` are supported, and `int8` only works for `llama`
+               , `gptneox` and `starcoder`.
+        :param cache_dir: (optional) This parameter will only be used when ``native`` is True
+               and ``pretrained_model_name_or_path`` is a huggingface checkpoint or hub repo id.
                It indicates the saving path for the converted low precision model.
-        :param tmp_path: (optional) Which path to store the intermediate fp16 model during the
-               conversion process. Default to `None` so that intermediate model will not be saved.
-        :param kwargs: keyword arguments which will be passed to the model instance
+        :param tmp_path: (optional) This parameter will only be used when ``native`` is True.
+               Which path to store the intermediate fp16 model during the conversion process.
+               Default to `None` so that intermediate model will not be saved.
```

Comment:
These two are not explicit in args?

Reply:
Right, they are not specified, and maybe we can remove them?